packages/block-editor/src/components/html-element-select-control/index.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { SelectControl, Notice } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '../../store';
+import { htmlElementMessages } from './messages';
+
+/**
+ * Renders a SelectControl for choosing HTML elements with validation
+ * to prevent duplicate <main> elements.
+ *
+ * @param {Object}   props          Component props.
+ * @param {string}   props.tagName  The current HTML tag name.
+ * @param {Function} props.onChange Function to call when the tag is changed.
+ * @param {string}   props.clientId The client ID of the current block.
+ * @param {Array}    props.options  SelectControl options (optional).
+ *
+ * @return {Component} The HTML element select control with validation.
+ */
+export default function HTMLElementSelectControl( {
```

Comment:
```suggestion
export default function HTMLElementControl( {
```

Suggestion: Maybe we should drop the `Select` part from the component name? There's no need to mention implementation details in the name.